src/back/spirv/parser.rs

```diff
@@ -0,0 +1,1545 @@
+/*! Standard Portable Intermediate Representation (SPIR-V) backend !*/
+use super::layout::{Instruction, Module};
+use crate::FastHashMap;
+use spirv::*;
+
+trait LookupHelper<T> {
+    type Target;
+    fn lookup_id(&self, handle: &crate::Handle<T>) -> Option<Word>;
+    fn lookup_handle(&self, word: &Word) -> Option<crate::Handle<T>>;
+}
+
+impl<T> LookupHelper<T> for FastHashMap<Word, crate::Handle<T>> {
+    type Target = T;
+
+    fn lookup_id(&self, handle: &crate::Handle<T>) -> Option<Word> {
+        let mut word = None;
+        for (k, v) in self.iter() {
+            if v.eq(handle) {
+                word = Some(*k);
+                break;
+            }
+        }
+        word
+    }
+
+    fn lookup_handle(&self, word: &u32) -> Option<crate::Handle<T>> {
+        let mut handle = None;
+        for (k, v) in self.iter() {
+            if k.eq(word) {
+                handle = Some(*v);
+                break;
+            }
+        }
+        handle
+    }
+}
+
+#[derive(Debug, PartialEq)]
+struct LookupFunctionType {
+    parameter_type_ids: Vec<Word>,
+    return_type_id: Word,
+}
+
+pub struct Parser {
+    module: Module,
+    id_count: u32,
+    capabilities: Vec<Capability>,
+    debugs: Vec<Instruction>,
+    annotations: Vec<Instruction>,
+    debug_enabled: bool,
+    void_type: Option<u32>,
```

Comment:
should this use `Word`

Reply:
Yes it should, and I will change it, but this should not be a field on its own.